src/prototype_room_creepbuilder.js

```diff
@@ -211,7 +211,14 @@ Room.prototype.getPartsStringDatas = function(parts, energyAvailable) {
 Room.prototype.getSettings = function(creep) {
   const role = creep.role;
   const updateSettings = roles[role].updateSettings && roles[role].updateSettings(this, creep);
-  const settings = _.merge(roles[role].settings, updateSettings);
+  if (config.debug.spawn) {
```

Comment:
Instead of having the `if clause` here, you can use the `debugLog`, like:
```this.debugLog('spawn', `getSettings updateSettings: ${JSON.stringify(updateSettings)}`);``` (checkout `logging.js`)
which reduces the complexity in this methods and has the same functionality.

Reply:
Oh I overlooked that I will change these accordingly.